### PR TITLE
Feature/80-static-css

### LINKED ISSUE #80: Rendering static styles
**Current state** - We use Fela's [renderStatic](https://fela.js.org/docs/latest/api/fela/renderer#renderstatic) to render 3rd party or app static styles. We also use [renderFont](https://fela.js.org/docs/latest/api/fela/renderer#renderfont) for rendering `@font-face`s

**Suggested state** - Change webpack config in `@ackee/react-scripts` to export static files as pure css and link them into the `index.html` so that they're loaded in parallel with page. Make the same for fonts. In all that cases we can utilize techniques like [prefetching or preloading](https://css-tricks.com/prefetching-preloading-prebrowsing/).

Todo
- [x] research how to lazy load non critical (usually 3rd party component related) CSS

### PR DESCRIPTION
**Pull request by jancama2**
# Updates
- `react-scripts`
   - extracting css to separate files in production (loaded via `style-loader` in dev)
   - support loading styles to a specific placeholder in `<head>` while lazy-loading to ensure ordering of styles and correct work together with `fela`
- `cra-template-typescript`
    - migrate code to work with the changes in scripts

Closes #80

### CODE REVIEW

**Review by filip-dot-cube — APPROVED** [review]

**Review by jstorm31 — APPROVED** [review]

File: packages/react-scripts/custom/config/utils/injectLazyPlaceholder.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  );
+
+  if (pluginIndex >= 0) {
+    plugins[pluginIndex] = new MiniCssExtractPlugin({
+      // start - These options are copied from official webpack.config.js
+      filename: 'static/css/[name].[contenthash:8].css',
+      chunkFilename: 'static/css/[name].[contenthash:8].chunk.css',
+      // end
+      insert: `#${LAZY_STYLES_PLACEHOLDER_ID}`,
+    });
```
Comment (cermakjiri) [inline packages/react-scripts/custom/config/utils/injectLazyPlaceholder.js:30]: Nice solution, better than modifying the original webpack.config.js directly 👍

**Comment by cermakjiri** [comment]
Looks awesome 👏   Could you please add [to release draft](https://github.com/AckeeCZ/create-react-app/releases/tag/untagged-57a71f3d88c2640278b5) what steps are required for the migration on projects with the current version of react-scripts (v2.x)? 🙏

**Review by cermakjiri — APPROVED** [review]

**Comment by jancama2** [comment]
@cermakjiri I've updated [the release draft](https://github.com/AckeeCZ/create-react-app/releases/edit/untagged-0a3fcce3368aa58bcf12) 👌 . Shouldn't we make a release for `cra-template-typescript` too?

**Comment by cermakjiri** [comment]
> @cermakjiri I've updated [the release draft](https://github.com/AckeeCZ/create-react-app/releases/edit/untagged-0a3fcce3368aa58bcf12) 👌 . Shouldn't we make a release for `cra-template-typescript` too?

Looks good, thanks ❤️ 
Sure, we should. But it'll be more presentation of updates rather than migration guide (i.e. `cra-template-typescript` is used only during project bootstrapping, it can't be upgraded such as `react-scripts`).

**Pull request merged**